onnx/defs/schema.h

```diff
@@ -242,27 +225,53 @@ class OpSchema {
 
   struct Attribute {
     Attribute(
-        const char* name_,
-        const char* description_,
-        AttributeProto::AttributeType type_,
-        bool required_)
+        const std::string& name_,
```

Comment:
Why don't we have a function sig like this:
Attr(name, description, type)
Attr(name, description, type, default_value)

I think type is helpful for reading and it's easy for the compiler to figure out what is going on.

Reply:
As discussed, changed to keep the type information and let's see whether there're other opinions. Thank you!